addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRuleUnitTest.java

```diff
@@ -362,4 +363,58 @@ void crossDomainScriptInContextLow() throws HttpMalformedHeaderException {
                 alertsRaised.get(0).getEvidence(),
                 equalTo("<script src=\"https://www.otherDomain.com/script2\"/>"));
     }
+
+    @Test
+    void dontRaiseIssueWhenLinkIsTrustedDomain() throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example2.com/test/ HTTP/1.1");
+        rule.getConfig()
+                .setProperty(RuleConfigParam.RULE_DOMAINS_TRUSTED, "https://www.example.com/.*");
+
+        msg.setResponseBody(
+                "<html>"
+                        + "<head>"
+                        + "<script src=\"https://www.example2.com/script1\"/>"
```

Comment:
This is the same domain as the message, should be using the one being trusted.

Reply:
I also added integrity in the `ResponseBody`, because without it, alert is being raised.

Reply:
That means the current condition is not correct, see previous comment.